### PR TITLE
SwatchColorPicker: Expose onRenderColorCell

### PR DESCRIPTION
**Pull request by bhavinkotak07**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [ ] Include a change request file using `$ yarn change`

#### Description of changes

(give an overview)
Reference PR: [14839](https://github.com/microsoft/fluentui/pull/14839/)
SwatchColorPicker doesn't have any onRender function. With this change we can customize the UI of the Color picker to show a color and font picker combo. Below is the screenshot which shows the proposed change.
image

#### Focus areas to test

(optional)

### CODE REVIEW

File: packages/react-next/src/components/SwatchColorPicker/SwatchColorPicker.base.tsx [hunk 1]
```diff
@@ -147,7 +147,10 @@ export class SwatchColorPickerBase extends React.Component<ISwatchColorPickerPro
       />
     );
   }
-
+  private _onRenderItem = (item: IColorCellProps, index: number): JSX.Element => {
+    const { onRenderColorCell = this._renderOption } = this.props;
+    return onRenderColorCell(item, this._renderOption) as JSX.Element;
```
Comment (RupamSarkar) [inline packages/react-next/src/components/SwatchColorPicker/SwatchColorPicker.base.tsx:152]: Do we need `as JSX.Element` here?
I think as we added `null` to the return type of `onRenderItem` in `Grid.types.ts` , we can add the `null` return type here as well ...

Comment (bhavinkotak07) [reply packages/react-next/src/components/SwatchColorPicker/SwatchColorPicker.base.tsx]: I have added null to the return type of _Grid.types.ts_ and have remove _as JSX.Element_

File: change/@fluentui-react-next-2020-09-02-10-05-14-master.json [hunk 2]
```diff
@@ -0,0 +1,8 @@
+{
+  "type": "prerelease",
```
Comment (ecraig12345) [inline change/@fluentui-react-next-2020-09-02-10-05-14-master.json:2]: Delete this file

File: change/office-ui-fabric-react-2020-09-01-11-04-49-master.json [hunk 3]
```diff
@@ -0,0 +1,8 @@
+{
+  "type": "minor",
+  "comment": "Expose onRenderColorItem and override",
```
Comment (ecraig12345) [inline change/office-ui-fabric-react-2020-09-01-11-04-49-master.json:3]: This goes in the release notes, so it's good to include which component is being modified.
```suggestion
  "comment": "SwatchColorPicker: Expose onRenderColorItem and override",
```

File: change/office-ui-fabric-react-2020-09-01-19-13-27-master.json [hunk 4]
```diff
@@ -0,0 +1,8 @@
+{
+  "type": "minor",
```
Comment (ecraig12345) [inline change/office-ui-fabric-react-2020-09-01-19-13-27-master.json:2]: Delete this file

**Review by ecraig12345 — COMMENTED** [review]
Thanks for porting the change! This mostly looks good, but I have a few minor comments about the change files. Also you'll need to merge with master to get rid of the change in the web-components package (that happened because the change was missing from master, and I just added it).

File: change/office-ui-fabric-react-2020-09-01-19-13-27-master.json (hunk 4 above)
Comment (bhavinkotak07) [reply change/office-ui-fabric-react-2020-09-01-19-13-27-master.json]: Deleted the requested file

File: change/@fluentui-react-next-2020-09-02-10-05-14-master.json (hunk 2 above)
Comment (bhavinkotak07) [reply change/@fluentui-react-next-2020-09-02-10-05-14-master.json]: Deleted the requested file

File: change/office-ui-fabric-react-2020-09-01-11-04-49-master.json (hunk 3 above)
Comment (bhavinkotak07) [reply change/office-ui-fabric-react-2020-09-01-11-04-49-master.json]: Thanks for the feedback. Updated the comment as requested

File: change/@fluentui-web-components-2020-09-24-11-27-41-bhkotak-swatch-update.json [hunk 5]
```diff
@@ -0,0 +1,8 @@
+{
+  "type": "none",
+  "comment": "Merge branch 'master' of https://github.com/bhavinkotak07/fluentui into bhkotak/swatch-update",
```
Comment (ecraig12345) [inline change/@fluentui-web-components-2020-09-24-11-27-41-bhkotak-swatch-update.json:3]: Delete this file, it shouldn't be needed

Comment (bhavinkotak07) [reply change/@fluentui-web-components-2020-09-24-11-27-41-bhkotak-swatch-update.json]: Done.

**Review by ecraig12345 — APPROVED** [review]
Thanks for doing this and making the updates!

**Pull request merged**